app/src/main/java/com/sci/recipeandroid/feature/auth/data/service/FacebookAuthenticator.kt

```diff
@@ -0,0 +1,45 @@
+package com.sci.recipeandroid.feature.auth.data.service
+
+import android.content.Intent
+import androidx.fragment.app.Fragment
+import com.facebook.AccessToken
+import com.facebook.CallbackManager
+import com.facebook.FacebookCallback
+import com.facebook.FacebookException
+import com.facebook.login.LoginManager
+import com.facebook.login.LoginResult
+
+class FacebookAuthenticator {
+
+    private val callbackManager: CallbackManager = CallbackManager.Factory.create()
+
+    fun handleFacebookActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        callbackManager.onActivityResult(requestCode, resultCode, data)
+    }
+
+    fun facebookLogin(
```

Comment:
as Backend need to pass token get from Facebook SDK , we need to generate first token , so this method name should be generateFaceBookToken . Becuz this is still generate token step , not yet reach to actual fb logic step.

Reply:
and google authenticator class too .